include/RAJA/policy/cuda/launch.hpp

```diff
@@ -116,13 +202,36 @@ void launch_global_fcn_fixed(BODY body_in)
   body(ctx);
 }
 
+template <typename BODY, int num_threads, size_t BLOCKS_PER_SM, typename ReduceParams>
+__global__ void launch_new_reduce_global_fcn_fixed(BODY body_in, ReduceParams reduce_params)
+{
+  LaunchContext ctx;
+
+  using RAJA::internal::thread_privatize;
+  auto privatizer = thread_privatize(body_in);
+  auto& body = privatizer.get_priv();
+
+  //Set pointer to shared memory
+  extern __shared__ char raja_shmem_ptr[];
+  ctx.shared_mem_ptr = raja_shmem_ptr;
+
+  RAJA::expt::invoke_body( reduce_params, body, ctx );
+
+  //Using a flatten global policy as we may use all dimensions
+  RAJA::expt::ParamMultiplexer::combine<RAJA::cuda_flatten_global_xyz_direct>(reduce_params);
```

Comment:
It would be nice to be able to pass the num_threads information in to combine. All we know here is a max number of threads right? We don't know the shape of the block or if the block uses all num_threads?

Reply:
We can make runtime calls to get the shape of the thread blocks if that helps.

Reply:
I'm interested in ways to add compile time knowledge, but we can think about this more later.